### PR TITLE
Updates documentation to filter()

### LINKED ISSUE #4791: Refresh `filter()` docs
* [ ] Proofread title, description, and parameters 
* [ ] Proofread comments in examples 
* [ ] Ensure comment headings used to divide into scannable sections 
* [ ] Ensure there's one example that shows tidy eval and links to `?dplyr_tidy_eval` 
* [ ] Ensure there's one example that shows `across()` 
* [ ] Flag examples that seem overly complex for further review 
* [ ] Switch out mtcars/iris for something more interesting, or a smaller dataset constructed specifically to illustrate one point. 
* [ ] Review size of example outputs to ensure that you can easily see important rows and cols. 
* [ ] Datasets should always be tibbles to avoid use of `as_tibble()` distracting from the main point

### PR DESCRIPTION
**Pull request by djnavarro**
Various updates to the filter() documentation. 

- Adds brief introduction to make the "details" section less jarring
- Minor tidying to text to improve readability
- Attempts to describe the arguments more clearly
- Removes "see alsos" that will be redundant in an `across()` workflow

Closes #4791. I hope :slightly_smiling_face:

### CODE REVIEW

File: R/filter.R [hunk 1]
```diff
@@ -1,15 +1,23 @@
-#' Subset rows using column values
+#' Return all rows that satisfy conditions
```
Comment (hadley) [inline R/filter.R]: Can you take a look at the headings on https://dplyr.tidyverse.org/dev/reference/index.html to try and make sure this matches the same pattern?

File: R/filter.R [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 #'
-#' `filter()` retains the rows where the conditions you provide a `TRUE`. Note
-#' that, unlike base subsetting with `[`, rows where the condition evaluates
-#' to `NA` are dropped.
+#' The `filter()` function is used to subset a data object,
```
Comment (hadley) [inline R/filter.R]: data frame, maybe?

File: R/filter.R [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+#' retaining all rows that satisfy user-specified conditions.
```
Comment (hadley) [inline R/filter.R]:
```suggestion
#' retaining all rows that satisfy your conditions.
```

File: R/filter.R [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+#' To be retained, the row must produce a value of `TRUE` for all conditions.
```
Comment (hadley) [inline R/filter.R]: Needs a segue into next sentence?

File: R/filter.R [hunk 5]
```diff
@@ -32,43 +40,48 @@
...
-#' average.
 #' @family single table verbs
 #' @inheritParams arrange
-#' @param ... <[`data-masking`][dplyr_data_masking]> Logical predicates defined
-#'   in terms of the variables in `.data`.
-#'   Multiple conditions are combined with `&`. Only rows where the
-#'   condition evaluates to `TRUE` are kept.
-#' @param .preserve when `FALSE` (the default), the grouping structure
-#'   is recalculated based on the resulting data, otherwise it is kept as is.
+#' @param ... Expressions that return a
```
Comment (hadley) [inline R/filter.R]: Can you keep the data-masking badge in the first position please? We're trying to make this a convention

File: R/filter.R [hunk 6]
```diff
@@ -32,43 +40,48 @@
...
+#'   logical value, and are defined in terms of the variables in `.data`.
+#'   If multiple expressions are included, they are combined with the `&` operator.
+#'   Only rows for which all conditions evaluate to `TRUE` are kept. Note that
+#'   <[`data-masking`][dplyr_data_masking]> applies here.
+#' @param .preserve Relevant when the `.data` input is grouped.
+#'   If `.preserve = FALSE` (the default), the grouping structure
+#'   is recalculated based on the resulting data, otherwise the grouping is kept as is.
 #' @return
-#' An object of the same type as `.data`.
+#' An object of the same type as `.data`. The output has the following properties:
```
Comment (hadley) [inline R/filter.R]: Can you please file an issue to apply this change to the documentation for all verbs?

**Comment by djnavarro** [comment]
I think this removes the relevant nits :-)

**Comment by hadley** [comment]
Much better, thank you!

**Pull request merged**